lib/chef/event_dispatch/dispatcher.rb

```diff
@@ -28,6 +28,9 @@ def register(subscriber)
       # Define a method that will be forwarded to all
       def self.def_forwarding_method(method_name)
         define_method(method_name) do |*args|
+          if method_name == :deprecation && args.size == 1
```

Comment:
My only issue there was copy/pastaing the large amount of stuff in the metaprogramming ... but I can just stick all that in a helper method in this case.